### PR TITLE
Fix BYOND membership status available to non-members

### PR DESCRIPTION
**Pull request by lessthnthree**
## About The Pull Request

Fixes an issue introduced by https://github.com/Skyrat-SS13/Skyrat-tg/pull/18946 where players on the server donator list are marked as BYOND membership holders.

This adds a variable, donator_status that correctly holds a player's donator status to the server and unlocks the associated content.

## How This Contributes To The Nova Sector Roleplay Experience

We're all thankful for the work Lummox puts in to keep BYOND alive and he doesn't ask for much. Circumventing the BYOND membership check in exchange for a donation to the server instead is hopefully not intended behavior.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image

</details>

## Changelog

:cl: LT3
fix: Fixed BYOND membership status being available to non-BYOND members
/:cl:

### CODE REVIEW

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
I'd rather we simply don't show the BYOND icon in OOC next to their name when people aren't BYOND members, rather than doing this, as it also limits a few other things that donators would otherwise have access to, such as different ghost sprites and orbit options.

I wasn't the one that originally gave the unlock_contents to donators, that was done before that player ranks stuff, it's just that I refactored it to be cleaner.

TL;DR: just make sure the only thing the non-BYOND-member donators lose is the BYOND icon in OOC, as I think currently this would also result in more perks being accidentally taken away.

**Comment by lessthnthree** [comment]
> as it also limits a few other things that donators would otherwise have access to, such as different ghost sprites and orbit options.

The things that are for BYOND premium members, yes.
 
> I wasn't the one that originally gave the unlock_contents to donators, that was done before that player ranks stuff, it's just that I refactored it to be cleaner.
> 
> TL;DR: just make sure the only thing the non-BYOND-member donators lose is the BYOND icon in OOC, as I think currently this would also result in more perks being accidentally taken away.

They're not accidentally being taken away. They were accidentally given to non members. It is morally bankrupt to be offering the BYOND membership perks to non members in exchange for donations to your own community.

If your codebase is fine with that, so be it. You've said it here. Others have already corrected it.

**Review by GoldenAlpharex — APPROVED** [review]
I've thought about this some more, and discussed it with vinyl, and I think you're right. Those perks were made available to donators by accident, I'd rather have them get perks intentionally moving forward. We can just look into getting them access to the OOC chat color pref again in the future, but that'll be in a different PR.

**Comment by lessthnthree** [comment]
Better to be right late than never.

**Comment by GoldenAlpharex** [comment]
Just updated the PR description to properly reflect the PR that added that change.
